experimental/pkg/apis/mci/v1alpha1/types.go

```diff
@@ -0,0 +1,117 @@
+// multiple clusters.
+// The API schema here is very similar to a single cluster Ingress resource
+// with the primary difference being that the service name referenced in the
+// path refers to a MultiClusterService resource rather than a Service object.
+// +k8s:openapi-gen=true
+type MultiClusterIngress struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec               MultiClusterIngressSpec   `json:"spec"`
+	Status             MultiClusterIngressStatus `json:"status,omitempty"`
+}
+
+// MultiClusterIngressSpec is the spec for a MultiClusterIngress.
+type MultiClusterIngressSpec struct {
+	// Template contains the ingress spec used to create a
+	// MultiClusterIngress.
+	Template IngressSpecTemplate `json:"template,omitempty"`
+}
+
+// MultiClusterIngressStatus is the status of a MultiClusterIngress.
+type MultiClusterIngressStatus struct {
+	// VIP is the virtual IP address of the load balancer.
+	VIP string
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// MultiClusterIngressList is a list of MultiClusterIngress resources
+type MultiClusterIngressList struct {
+	meta_v1.TypeMeta `json:",inline"`
+	meta_v1.ListMeta `json:"metadata"`
+	Items            []MultiClusterIngress `json:"items"`
+}
+
+// IngressSpecTemplate contains the spec for MultiClusterIngress resource.
+// +k8s:openapi-gen=true
+type IngressSpecTemplate struct {
+	// +optional
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec               extensions.IngressSpec `json:"spec"`
+}
+
+// MultiClusterService is the API resource used to create services in multiple
+// clusters.
+// This is replicated across multiple clusters.
+// By doing cluster selection with Service instead of with Ingress, we can have
+// setups where different paths on the same MultiClusterIngress can route
+// traffic to multiple clusters.
+// For example, /foo can route traffic to clusters A and B and /bar can
+// route traffic to clusters B and C for a single MultiClusterIngress.
+// +k8s:openapi-gen=true
+type MultiClusterService struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec               MultiClusterServiceSpec   `json:"spec"`
+	Status             MultiClusterServiceStatus `json:"status,omitempty"`
+}
+
+// MultiClusterServiceSpec is the spec for a MultiClusterService.
+type MultiClusterServiceSpec struct {
+	// Spec for the service that will be created in individual clusters.
+	Template ServiceSpecTemplate `json:"template,omitempty"`
+	// Clusters selects the list of target clusters where the Service resource
+	// should be replicated.
+	Clusters []MultiClusterDestination `json:"clusters,omitempty"`
+}
+
+// MultiClusterServiceStatus is the status of a MultiClusterService.
+type MultiClusterServiceStatus struct{}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
As with line 52, it seems to me that this is causing linting to fail.

./experimental/pkg/apis/mci/v1alpha1/types.go:96:1: comment on exported type MultiClusterServiceList should be of the form "MultiClusterServiceList ..." (with optional leading article)

https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/GoogleCloudPlatform_k8s-multicluster-ingress/212/pull-kubernetes-multicluster-ingress-test/275/